File: src/php/Scimark2/Scimark2Benchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Scimark2 {

    use EntityFX\NetBenchmark\Core\Writer;
    use EntityFX\NetBenchmark\Core\Generic\BenchmarkBase;

    class Scimark2Benchmark extends BenchmarkBase {

        private $dhrystone;

        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Ratio = 300;
            $this->dhrystone = new Scimark2($printToConsole);
        }

        public function Warmup($aspect = 0.05) {
        }

        protected function BenchImplementation() {
            return $this->dhrystone->bench();
        }

        protected function PopulateResult($benchResult, $result) {
            $benchResult["Points"] = $result["CompositeScore"] * $this->Ratio;
            $benchResult["Result"] = $result["CompositeScore"];
            $benchResult["Units"] = "CompositeScore";
            $benchResult["Output"] = $result["Output"];
            return $benchResult;
        }
    }
}

File: src/lua/scimark2/scimark2Benchmark.lua
require "scimark2/scimark2"

Scimark2Benchmark = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.name = "Scimark2Benchmark"
    a.ratio = 10
    a.scimark2 = Scimark2(printToConsole)
end)

function Scimark2Benchmark:benchImplementation()
    return self.scimark2:bench(false)
end

function Scimark2Benchmark:warmup(aspect)

end

function Scimark2Benchmark:populateResult(benchResult, result)
    benchResult.points = result.CompositeScore * self.ratio
    benchResult.result = result.CompositeScore
    benchResult.units = "CompositeScore"
    benchResult.output = result.Output
    return benchResult
end